Modernize Objective-C Strings

LumioseSil opened the pull request:
#### 2fe6e434a10b8a482998d1b512b122be1559ace8
<pre>
Modernize Objective-C Strings
<a href="https://bugs.webkit.org/show_bug.cgi?id=251694">https://bugs.webkit.org/show_bug.cgi?id=251694</a>

Reviewed by NOBODY (OOPS!).

There are places where we can replace the stringWithUTFString method with the
shorthand version of @. This makes the code more clear and concise.

*Source\JavaScriptCore\API\JSWrapperMap.mm:
*Source\JavaScriptCore\testmem\testmem.mm:
*Source\WebCore\accessibility\mac\WebAccessibilityObjectWrapperMac.mm:
*Source\WebCore\platform\audio\mac\AudioBusMac.mm:
*Source\WebCore\platform\graphics\mac\ImageMac.mm:
*Source\WebCore\platform\ios\WebAVPlayerController.mm:
*Source\WebKit\UIProcess\WebsiteData\Cocoa\WebsiteDataStoreCocoa.mm:
*Source\WebKit\WebProcess\cocoa\WebProcessCocoa.mm:
*Source\WebKitLegacy\mac\Misc\WebCoreStatistics.mm:
*Source\WebKitLegacy\mac\Misc\WebLocalizableStrings.mm:
*Tools\DumpRenderTree\mac\DumpRenderTree.mm:
*Tools\DumpRenderTree\mac\ResourceLoadDelegate.mm:
*Tools\DumpRenderTree\mac\UIDelegate.mm:
*Tools\TestWebKitAPI\Tests\WebKitCocoa\FTP.mm:
*Tools\TestWebKitAPI\Tests\WebKitCocoa\ShouldOpenExternalURLsInNewWindowActions.mm:
*Tools\TestWebKitAPI\Tests\WebKitCocoa\WKURLSchemeHandler-1.mm:
*Tools\TestWebKitAPI\Tests\WebKitCocoa\WebPushDaemon.mm:
*Tools\TestWebKitAPI\cocoa\DaemonTestUtilities.mm:
*Tools\TestWebKitAPI\cocoa\PlatformUtilitiesCocoa.mm:
*Tools\TestWebKitAPI\mac\JavaScriptTestMac.mm:
*Tools\TestWebKitAPI\mac\PlatformUtilitiesMac.mm:
*Tools\WebKitTestRunner\cocoa\TestControllerCocoa.mm:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e5920b0d2d5c0fca430db1548567c11b859dcb9c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107371 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16431 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40293 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116530 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115957 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111271 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17854 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/7670 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99532 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113135 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13467 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96665 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41093 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95371 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28291 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/82857 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/96716 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9443 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29644 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/96141 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/7439 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10097 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6521 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/30781 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15599 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49225 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/104981 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11654 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26027 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Tools/DumpRenderTree/mac/DumpRenderTree.mm:
```diff
@@ -133,7 +133,7 @@
 static RetainPtr<NSString> toNS(const std::string& string)
 {
-    return adoptNS([[NSString alloc] initWithUTF8String:string.c_str()]);
+    return adoptNS(@(string.c_str()]));
```
I believe this will cause an overrelease. We should either keep it the way it currently is, or just `return @(string.c_str())` and let it implicitly initialize a `RetainPtr`.

whsieh commented on Tools/TestWebKitAPI/cocoa/DaemonTestUtilities.mm:
```diff
@@ -56,7 +56,7 @@ - (void)waitUntilExit;
     _NSGetExecutablePath(buffer.data(), &size);
     buffer[size] = '\0';
-    auto pathString = adoptNS([[NSString alloc] initWithUTF8String:buffer.data()]);
+    auto pathString = adoptNS(@(buffer.data());
```
Same here — this will trigger an overrelease.

whsieh reviewed (commented):
Is there a corresponding Bugzilla bug? If not, we should file one and include it in the commit message (where it currently says `Need the bug URL (OOPS!).`).

The commit message is also missing the file-by-file changes section.

cdumez commented on Source/WebCore/platform/mac/PlatformPasteboardMac.mm:
```diff
@@ -375,7 +375,7 @@ static String typeIdentifierForPasteboardType(const String& pasteboardType)
 
     NSArray *urlWithTitle = @[ @[ urlString ], @[ pasteboardURL.title ] ];
-    NSString *pasteboardType = [NSString stringWithUTF8String:WebURLsWithTitlesPboardType];
+    NSString *pasteboardType = [NSString initWithCString:WebURLsWithTitlesPboardType encoding:NSASCIIStringEncoding];
```
Isn't this missing a `[NSString alloc]`? Also, seems we should adoptNS() to avoid leaking now.

cdumez replied:
Maybe you meant to use https://developer.apple.com/documentation/foundation/nsstring/1497310-stringwithcstring?language=objc ?

cdumez reviewed (commented):
I didn't see any obvious issues in this latest iteration but I'll let someone with a bit more ObjC knowledge do the formal review.

aproskuryakov commented on Source/WebCore/platform/mac/PlatformPasteboardMac.mm:
```diff
@@ -375,7 +375,7 @@ static String typeIdentifierForPasteboardType(const String& pasteboardType)
 
     NSArray *urlWithTitle = @[ @[ urlString ], @[ pasteboardURL.title ] ];
-    NSString *pasteboardType = [NSString stringWithUTF8String:WebURLsWithTitlesPboardType];
+    NSString *pasteboardType = [NSString stringWithCString:WebURLsWithTitlesPboardType encoding:NSASCIIStringEncoding];
```
Some changes in this PR are just syntax, and others are a little harder to reason about. It would be best to either split it into mechanical and non-mechanical parts, or to have specific explanations in the commit message.

For example, this one probably reduces efficiency (more argument to pass, and I'm not sure if conversion from NSASCIIStringEncoding to UTF-8 is actually free, there are often edge cases to deal with unexpected input).

AZero13 replied:
I mean PBOARDTYPE is literally an ASCIILiteral type

aproskuryakov replied:
I understand that the string is ASCII only, however this doesn't necessarily mean that using `NSASCIIStringEncoding` to convert it into NSString is higher performing or more correct in any way. I suspect that it's the opposite.

aproskuryakov commented on Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:
```diff
@@ -155,8 +155,7 @@ static bool experimentalFeatureEnabled(const String& key, bool defaultValue = fa
 #if !RELEASE_LOG_DISABLED
-    static NSString * const WebKitLogCookieInformationDefaultsKey = @"WebKitLogCookieInformation";
-    shouldLogCookieInformation = [defaults boolForKey:WebKitLogCookieInformationDefaultsKey];
+    shouldLogCookieInformation = [defaults boolForKey:@"WebKitLogCookieInformation"];
```
Does this notation create a static string, or a new one each time the function is entered?

AZero13 replied:
Should be static. 

C compilers make a read only copy and the pointers are passed.

I imagine NSString literals are just Plain Old Data

LumioseSil committed:
Modernize Objective-C Strings
https://bugs.webkit.org/show_bug.cgi?id=251694

Reviewed by NOBODY (OOPS!).

There are places where we can replace the stringWithUTFString method with the
shorthand version of @. This makes the code more clear and concise.

*Source\JavaScriptCore\API\JSWrapperMap.mm:
*Source\JavaScriptCore\testmem\testmem.mm:
*Source\WebCore\accessibility\mac\WebAccessibilityObjectWrapperMac.mm:
*Source\WebCore\platform\audio\mac\AudioBusMac.mm:
*Source\WebCore\platform\graphics\mac\ImageMac.mm:
*Source\WebCore\platform\ios\WebAVPlayerController.mm:
*Source\WebKit\UIProcess\WebsiteData\Cocoa\WebsiteDataStoreCocoa.mm:
*Source\WebKit\WebProcess\cocoa\WebProcessCocoa.mm:
*Source\WebKitLegacy\mac\Misc\WebCoreStatistics.mm:
*Source\WebKitLegacy\mac\Misc\WebLocalizableStrings.mm:
*Tools\DumpRenderTree\mac\DumpRenderTree.mm:
*Tools\DumpRenderTree\mac\ResourceLoadDelegate.mm:
*Tools\DumpRenderTree\mac\UIDelegate.mm:
*Tools\TestWebKitAPI\Tests\WebKitCocoa\FTP.mm:
*Tools\TestWebKitAPI\Tests\WebKitCocoa\ShouldOpenExternalURLsInNewWindowActions.mm:
*Tools\TestWebKitAPI\Tests\WebKitCocoa\WKURLSchemeHandler-1.mm:
*Tools\TestWebKitAPI\Tests\WebKitCocoa\WebPushDaemon.mm:
*Tools\TestWebKitAPI\cocoa\DaemonTestUtilities.mm:
*Tools\TestWebKitAPI\cocoa\PlatformUtilitiesCocoa.mm:
*Tools\TestWebKitAPI\mac\JavaScriptTestMac.mm:
*Tools\TestWebKitAPI\mac\PlatformUtilitiesMac.mm:
*Tools\WebKitTestRunner\cocoa\TestControllerCocoa.mm:
